Tasks/KubernetesManifestV0/src/utils/DeploymentHelper.ts

```diff
@@ -263,6 +279,40 @@ function isPodReady(podStatus: any): boolean {
     return allContainersAreReady;
 }
 
+function getService(kubectl: Kubectl, serviceName) {
+    const serviceResult = kubectl.getResource(constants.DiscoveryAndLoadBalancerResource.service, serviceName);
+    utils.checkForErrors([serviceResult]);
+    return JSON.parse(serviceResult.stdout)
+}
+
+async function waitForServiceExternalIPAssignment(kubectl: Kubectl, serviceName: string): Promise<void> {
+    const sleepTimeout = 10 * 1000; // 10 seconds
+    const iterations = 18; // 18 * 10 seconds timeout = 3 minutes max timeout
+
+    for (let i = 0; i < iterations; i++) {
+        console.log(tl.loc('waitForServiceIpAssignment', serviceName));
+        await sleep(sleepTimeout);
+        let status = getService(kubectl, serviceName).status;
+        if (isLoadBalancerIPAssigned(status)) {
+            return;
+        }
+    }
+    tl.warning(tl.loc('waitForServiceIpAssignmentTimedOut', serviceName));
+}
+
+function isLoadBalancerIPAssigned(status: any) {
+    if (status.loadBalancer && status.loadBalancer.ingress && status.loadBalancer.ingress.length > 0) {
+        return true;
+    }
+    return false;
+}
+
+function getIngressResources(kubectl: Kubectl, ingressResources: Resource[]) {
```

Comment:
[NIT] This can be called above. Doesn't need an extra function. Given that it's a one liner.

Reply:
done